### ISSUE TITLE
Cannot use  down to expand ToolStripDropDownButton in toolStrip2 after using right/Left to collapse ToolStripDropDownButton in toolStrip1

### ISSUE THREAD

**Issue by Zheng-Li01** [dotnet/winforms#12916]
### .NET version

.NET SDK 10.0.100-preview2.25110.18

### Did it work in .NET Framework?

No

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

Can reproduce on all previous, NET versions

### Issue description

Cannot use down to expand ToolStripDropDownButton in toolStrip2 after using right/Left to collapse ToolStripDropDownButton in toolStrip1 

### Steps to reproduce

1. Create Winforms .NET app with 2 toolStrips
2. In these 2 toolStrips, add ToolStripDropDownButton, add toolStripMenuItem for ToolStripDropDownButton separately 
3. Set these 2 toolStrips' TabStop property to True 
4. Build and run app 
5. Press down to expand ToolStripDropDownButton in toolStrip1 - it will focus on toolStripMenuItem 
6. Press right/Left to collapse ToolStripDropDownButton in toolStrip1- it will focus on ToolStripDropDownButton  
7.  Tab to toolstrip2 button 
8. Press down to expand ToolStripDropDownButton in toolStrip2  -- it will focus on ToolStripDropDownButton in toolStrip1 

Or using the application & repeat the steps 4~8
![Image](https://github.com/user-attachments/assets/0c9769fe-2835-4ba9-8c95-ed8b35845e2c)

**More info:** 
1. The issue also can reproduce when using Enter and right keyboard combination 
![Image](https://github.com/user-attachments/assets/5bbb8368-3e48-45e4-a3e1-153c35d474de)
2. The issue cannot reproduce when using Enter & Enter or Enter & Alt or Down & Alt keyboard combinations

**Comment by John-Qiao** [comment]
Verified the issue on latest .NET SDK 10.0.100-preview.3.25155.10 + private dlls built from the main branch of WinForms repo. The issue has been fixed that we can use down to expand ToolStripDropDownButton in toolStrip2 after using right/Left to collapse ToolStripDropDownButton in toolStrip1.

https://github.com/user-attachments/assets/533482bc-b7c8-43d7-9a66-107c60cfe0b5

**Comment by Liv-Goh** [comment]
Verified with .NET 10.0 Preview 3 SDK test pass build, the issue has been fixed, and the testing result is same as above.
